src/acpi/acpi.c

```diff
@@ -1166,6 +1166,30 @@ unsigned long acpi_gtdt_add_watchdog(unsigned long current, uint64_t refresh_fra
 	return current + sizeof(struct acpi_gtdt_watchdog);
 }
 
+__weak unsigned long acpi_soc_fill_wdat(acpi_wdat_t *wdat, unsigned long current)
+{
+	return 0;
+}
+
+static void acpi_create_wdat(acpi_header_t *header, void *unused)
+{
+	if (!CONFIG(ACPI_WDAT_WDT))
```

Comment:
Should we create the table when watchdog is disabled?

Reply:
If neither `CONFIG_SOC_INTEL_COMMON_OC_WDT_WDAT` nor `SOC_INTEL_COMMON_BLOCK_TCO_WDT_WDAT` are selected then we should probably not generate the table, yes. I guess `ACPI_WDAT_WDT` should depend on a backend WDT being selected for it?

Reply:
@mkopec I meant UEFI variable

Reply:
Hmm, in that case I think we should, the firmware won't start the watchdog, but the OS can still decide to use it

Reply:
>  the OS can still decide to use it

What about user decision?

Reply:
I meant that the user can configure the OS to use or not to use the watchdog :) I don't see how hiding the WDAT table from the OS is useful

Reply:
Okay, if it is possible then all good.